docs/related-tools/local-analysis/running-spotbugs.md

```diff
@@ -0,0 +1,67 @@
+# Running SpotBugs
+
+To run SpotBugs as a [client-side tool](client-side-tools.md):
+
+1.  Enable **Run analysis through build server** in your repository **Settings**, tab **General**, **Repository analysis**.
+
+    This setting enables Codacy to wait for the results of the local analysis before resuming the analysis of your commits.
+
+    ![Run analysis through build server](images/run-analysis-through-build-server.png)
+
+1.  Obtain a [project API Token](../../repositories-configure/integrations/project-api.md) for your repository.
+
+    You need the project API Token to allow the Codacy Analysis CLI to authenticate to Codacy when reporting the analysis results.
+
+1.  Set the following environment variable to specify your project API Token:
+
+    ```bash
+    export CODACY_PROJECT_TOKEN=<your project API Token>
+    ```
+
+    !!! warning
+        **Never write API Tokens on your configuration files** and keep your API Tokens well protected, as they grant owner permissions to your projects.
+
+        We recommend that you set the API Tokens as environment variables. Check the documentation of your CI/CD platform on how to do this.
+
+1.  **If you are using Codacy Self-hosted** set the following environment variable to specify your Codacy instance URL:
+
+    ```bash
+    export CODACY_API_BASE_URL=<your Codacy instance URL>
+    ```
+
+1.  Compile your Java or Scala repository on your build server, as you would normally do.
+
+1.  Download and run the [Codacy Analysis CLI](https://github.com/codacy/codacy-analysis-cli#install){: target="_blank"} on the root of the repository, specifying the tool SpotBugs.
+
+    ```bash
+    codacy-analysis-cli analyze --tool spotbugs \
+                                --allow-network \
+                                --upload \
+                                --verbose
+    ```
+
+The Codacy Analysis CLI runs SpotBugs on the compiled classes of your repository and uploads the results to Codacy so you can use them in your workflow.
+
+## Detecting sources and compiled classes
+
+The Codacy Analysis CLI tries to find the compiled classes and map results to the source files automatically. If you use Maven, Gradle, or sbt the Codacy Analysis CLI also detects the default layouts automatically.
+
+If there is an issue with detection, you can configure these paths manually by adding a `.codacy.yml` [Codacy configuration file](../../repositories-configure/codacy-configuration-file.md) to the root of the repository:
+
+```yml
+---
+engines:
+  spotbugs:
+    enabled: true
```

Comment:
We have to remove this line. We don't support `enabled` for engines.

Reply:
Thanks @franciscodua, I just removed the line with `enabled: true`. Please check.